### ISSUE TITLE
Embedded site.py emits « DeprecationWarning: 'U' mode is deprecated »  with python 3.5.4 and 3.6.4

### ISSUE THREAD

**Issue by rhertzog** [pypa/virtualenv#1120]
I often run my unit tests with warnings enabled to ensure that my code is future-proof and since a few weeks (likely since my Debian sid system switched to python 3.5.4 and 3.6.4) those tests are failing with errors like this one:
```
  File "/home/rhertzog/deb/core/distro-tracker/.tox/py36-django111-no-warnings/lib/python3.6/site.py", line 165, in addpackage
    f = open(fullname, "rU")
DeprecationWarning: 'U' mode is deprecated
```

The site.py provided in python3.6 does not have this flag, hence I tracked this down to the site.py embedded in virtualenv (on which tox relies to run the tests in a isolated manner).

**Comment by mjpieters** [comment]
Even worse, you can't run in 'error' mode when using a virtualenv:

```
$ bin/python3.7 -W error - c '[(yield i) for i in range(3)]'  # demonstrate deprecation turned into syntax error in 3.7
Traceback (most recent call last):
  File ".../python-3.7/lib/python3.7/site.py", line 703, in <module>
    main()
  File ".../python-3.7/lib/python3.7/site.py", line 686, in main
    paths_in_sys = virtual_addsitepackages(paths_in_sys)
  File ".../python-3.7/lib/python3.7/site.py", line 644, in virtual_addsitepackages
    return addsitepackages(known_paths, sys_prefix=sys.real_prefix)
  File ".../python-3.7/lib/python3.7/site.py", line 282, in addsitepackages
    addsitedir(sitedir, known_paths)
  File ".../python-3.7/lib/python3.7/site.py", line 204, in addsitedir
    addpackage(sitedir, name, known_paths)
  File ".../python-3.7/lib/python3.7/site.py", line 165, in addpackage
    f = open(fullname, "rU")
DeprecationWarning: 'U' mode is deprecated
```

You have to enable warnings in the main script instead, which means syntax issues won't be caught, you have to use `exec` instead:

```
$ bin/python3.7 -c "import warnings; warnings.simplefilter('error');exec('''[(yield i) for i in range(3)]''')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1
SyntaxError: 'yield' inside list comprehension
```

**Comment by mjpieters** [comment]
I note that Python itself fixed this for Python 3.1, in 2011.

**Comment by gaborbernat** [comment]
fixed via #1064
